src/scripts/components/TimeEntriesList.tsx

```diff
@@ -93,41 +93,100 @@ type TimeEntriesListItemProps = {
   project: Toggl.Project | null;
 };
 function TimeEntriesListItem ({ timeEntries, project }: TimeEntriesListItemProps) {
+  const [groupEntries, setGroupEntries] = React.useState(true)
   const timeEntry = timeEntries[0];
-  const description = timeEntry.description || NO_DESCRIPTION;
-  const isBillable = !!timeEntry.billable;
-  const tags = (timeEntry.tags && timeEntry.tags.length > 0)
-    ? timeEntry.tags.join(', ')
-    : '';
 
   const totalDuration = timeEntries.reduce((sum, entry) => {
     if (entry.duration > 0) sum += entry.duration;
     return sum;
   }, 0);
   const entriesCount = timeEntries.length;
   const earliestStartTime = format(timeEntries[timeEntries.length - 1].start, 'HH:mm');
+  const isGrouped = entriesCount > 1 && groupEntries
+  const toggleGrouping = () => setGroupEntries(val => !val);
+
+  const content = [
+    entriesCount > 1 && (
+      <TimeEntryRow
+        renderHeader
+        key={`grouped-${timeEntry.id}`}
+        timeEntry={timeEntry}
+        onRowClick={toggleGrouping}
+        project={project}
+        entriesCount={entriesCount}
+        earliestStartTime={earliestStartTime}
+        totalDuration={totalDuration}
+      />
+    )
+  ].concat(
+    (!isGrouped ? timeEntries : []).map(timeEntry => (
+      <TimeEntryRow
+        renderHeader={false}
+        key={timeEntry.id}
+        timeEntry={timeEntry}
+        onRowClick={editEntry(timeEntry)}
+        project={project}
+      />
+    ))
+  );
 
   return (
-    <EntryItem>
+    <React.Fragment>
+      {content}
+    </React.Fragment>
+  );
+}
+
+function TimeEntryRow ({
+  timeEntry,
+  onRowClick,
+  project,
+  renderHeader,
+  entriesCount,
+  earliestStartTime,
+  totalDuration
+}: {
+  timeEntry: Toggl.TimeEntry,
+  onRowClick: (event: React.MouseEvent) => void,
+  project: Toggl.Project | null,
+  renderHeader: boolean,
+  entriesCount?: number,
+  earliestStartTime?: string,
+  totalDuration?: number,
+}) {
+
+  const description = timeEntry.description || NO_DESCRIPTION;
+  const isBillable = !!timeEntry.billable;
+  const tags = (timeEntry.tags && timeEntry.tags.length > 0)
+    ? timeEntry.tags.join(', ')
+    : '';
+
+  return (
+    <EntryItem onClick={onRowClick}>
       <EntryItemRow>
-        {entriesCount > 1 &&
+        {renderHeader && entriesCount && earliestStartTime &&
```

Comment:
would be nice if all the conditions around grouping were a bit more concise, but not a big deal

Reply:
It's just `renderHeader` that matters but I do an existential check here
so in the next line TS doesn't complain about optional properties